src/gazebo_uuv_plugin.cpp

```diff
@@ -39,8 +39,43 @@ void GazeboUUVPlugin::InitializeParams() {}
 
 void GazeboUUVPlugin::Publish() {}
 
+void GazeboUUVPlugin::ParseBuoyancy(physics::ModelPtr _model) {
+  physics::LinkPtr link_ptr;
```

Comment:
Shouldn't the `link_ptr` be declared inside the loop?

Reply:
Reimplemented the tag parsing.